core/src/main/java/org/mule/api/security/tls/TlsConfiguration.java

```diff
@@ -311,38 +320,40 @@ protected void checkKeyStoreContainsAlias(KeyStore keyStore) throws KeyStoreExce
 
     private void initTrustManagerFactory() throws CreateException
     {
-        if (null != trustStoreName)
+        if (null == trustStoreName && revocationCheck == null)
         {
-            Boolean revocationEnabled = revocationCheck != null;
+            return;
+        }
 
-            // Revocation checking is only supported for PKIX algorithm
-            if (revocationEnabled && !REVOCATION_KEYSTORE_ALGORITHM.equalsIgnoreCase(trustManagerAlgorithm))
-            {
-                String errorText = formatInvalidCrlAlgorithm(getTrustManagerAlgorithm());
-                throw new CreateException(CoreMessages.createStaticMessage(errorText), this);
-            }
+        Boolean revocationEnabled = revocationCheck != null;
 
-            try
-            {
-                KeyStore trustStore = createTrustStore();
-                trustManagerFactory = TrustManagerFactory.getInstance(trustManagerAlgorithm);
+        // Revocation checking is only supported for PKIX algorithm
+        if (revocationEnabled && !REVOCATION_KEYSTORE_ALGORITHM.equalsIgnoreCase(trustManagerAlgorithm))
+        {
+            String errorText = formatInvalidCrlAlgorithm(getTrustManagerAlgorithm());
+            throw new CreateException(CoreMessages.createStaticMessage(errorText), this);
+        }
 
-                if (revocationEnabled)
-                {
-                    ManagerFactoryParameters tmfParams = revocationCheck.configFor(trustStore);
-                    trustManagerFactory.init(tmfParams);
-                }
-                else
-                {
-                    trustManagerFactory.init(trustStore);
-                }
+        try
+        {
+            KeyStore trustStore = trustStoreName != null ? createTrustStore() : null;
```

Comment:
How could the trustStore be null? Wasn't there a check above?

Reply:
Yes. The check verifies that both trust store and revocation are null. If revocation is not null but trust store is, we must construct a custom trust manager for revocation with default java trust store.